firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Browser/SearchEngines/SearchEngineSelectionMiddlewareTests.swift

```diff
@@ -8,41 +8,61 @@ import XCTest
 @testable import Client
 
 final class SearchEngineSelectionMiddlewareTests: XCTestCase {
+    var mockStore: MockStoreForMiddleware<AppState>!
     var mockProfile: MockProfile!
-    var mockSearchEngines: [OpenSearchEngine]!
+    var mockSearchEnginesManager: SearchEnginesManager!
```

Comment:
nit: should we avoid `!` since we agreed on the swiftlint rules?

Since we are testing the middleware and not search engines manager, what do you think about creating + using the mock instead. I created a `MockSearchEnginesManager` in my PR: https://github.com/mozilla-mobile/firefox-ios/pull/23064, that can be used instead of creating the concrete class here.

Reply:
Does that rule specifically apply to this situation? 🤔  Because we do this pattern across _many_ tests (just search for "var mock" and see how many lines come up with `!` haha). If possible, I would propose removing that rule from the unit test target as a crash in unit tests isn't going to impact our users, but makes our tests more readable when dealing with mocks...

Oh you already made a mock for SearchEnginesManager? Wonderful! There are several places we should be using this mock in our current tests I believe. Maybe we also want to use it in the MockProfile, since the real search engine manager in there is also used across multiple tests (which is why I thought mocking `SearchEnginesManager` was going to be a larger task... we may want to make a separate ticket for those changes). Thanks @cyndichin! 😁

Reply:
Pushed up extended implementation to your mock, and usage in these tests 👍

Reply:
awesome! thanks for updating that! The `MockProfile` is another beast to tackle haha, I don't like how its using databases and would like to have that refactored, but I believe there is a profile tech debt project so hopefully it addresses that. I'll say we can make enhancements for future tests and leave old tests as-is.

Reply:
> I was under the assumption, but let me clarify with the team since I can see a lot of people not wanting this for unit tests. I'm not sure if we're doing this to any other rules though. In that PR, I also avoid using `!`, but it's still readable (at least to me 🤣 ), let me know what you think about that alternative!

I was always a dissenter on the ! rule. 😂 

It sort of depends on the test in question; sometimes it's annoying because we might need to unwrap the optional in certain situations (with a guard similar to the one you flagged to make simpler). Let's take it to the team like you suggested!